app/scripts/modules/core/cluster/task.matcher.ts

```diff
@@ -0,0 +1,92 @@
+import {has, intersection, map} from 'lodash';
+
+import {ITask} from '../task/task.read.service';
+import {ServerGroup} from '../domain/serverGroup';
+
+export interface ITaskMatcher {
+ (task: ITask, serverGroup: ServerGroup): boolean;
+}
+
+function createcopylastasgMatcher(task: ITask, serverGroup: ServerGroup): boolean {
+  const source = task.getValueFor('source'),
+        targetAccount: string = task.getValueFor('deploy.account.name'),
+        targetRegion: string = task.getValueFor('availabilityZones') ? Object.keys(task.getValueFor('availabilityZones'))[0] : null,
+        deployedServerGroups: {[region: string]: string[]} = task.getValueFor('deploy.server.groups'),
+        targetServerGroup: string = targetRegion && deployedServerGroups && deployedServerGroups[targetRegion] ? deployedServerGroups[targetRegion][0] : null,
+        sourceServerGroup: string = source.asgName,
+        sourceAccount: string = source.account,
+        sourceRegion: string = source.region;
+
+  if (serverGroup.account === targetAccount && serverGroup.region === targetRegion && serverGroup.name === targetServerGroup) {
+    return true;
+  }
+  return serverGroup.account === sourceAccount && serverGroup.region === sourceRegion && serverGroup.name === sourceServerGroup;
```

Comment:
Pretty hard to follow. Combine these two returns, confusing having them separate. Or better yet, store results in descriptive consts and return the `||` of those. Something like:
```
const serverGroupInTarget = serverGroup.account === targetAccount && serverGroup.region === targetRegion && serverGroup.name === targetServerGroup;
const serverGroupInSource = serverGroup.account === sourceAccount && serverGroup.region === sourceRegion && serverGroup.name === sourceServerGroup;
return serverGroupInTarget || serverGroupInSource;
```